### PR TITLE
Insight wrapper

### PR DESCRIPTION
**Pull request by hightower8083**
Added a method to read insight data from HDF5 file generated by the SourceLab intreface. Based on the generic FromArray profie

### CODE REVIEW

**Comment by hightower8083** [comment]
I think this one is ready for reviewing/merging @MaxThevenet

File: lasy/profiles/from_insight_file.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+class FromInsightFile(FromArrayProfile):
+    r"""
+    Profile defined from insight measurement.
+
+    Parameters
+    ----------
+    file_path: string
+        Path to the file created by insight which contains full field (e.g. Exyt_0.h5)
+
+    pol : list of 2 complex numbers (dimensionless)
```
Comment (MaxThevenet) [inline lasy/profiles/from_insight_file.py]:
```suggestion
    pol: list of 2 complex numbers (dimensionless)
```

File: lasy/profiles/from_insight_file.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+class FromInsightFile(FromArrayProfile):
+    r"""
+    Profile defined from insight measurement.
+
+    Parameters
+    ----------
+    file_path: string
+        Path to the file created by insight which contains full field (e.g. Exyt_0.h5)
```
Comment (MaxThevenet) [inline lasy/profiles/from_insight_file.py]:
```suggestion
        Path to the file created by INSIGHT that contains the full field (e.g. Exyt_0.h5)
```

File: lasy/profiles/from_insight_file.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        Polarization vector. It corresponds to :math:`p_u` in the above
+        formula ; :math:`p_x` is the first element of the list and
+        :math:`p_y` is the second element of the list. Using complex
+        numbers enables elliptical polarizations.
+
+    omega0: string or float
+        Set central frequency for the envelope constrauction. Can be a float value
+        in [1/s], or a string defining the method for automatic frequency detection:
+        "barycenter" frequency is averaged over the power profile, "peak" frequency
+        corresponding to the peak field.
```
Comment (MaxThevenet) [inline lasy/profiles/from_insight_file.py]: Minor typos fixed, could you check that this is still correct?
```suggestion
    omega0: string or float
        Set the central frequency for the envelope construction. Can be a float value
        in [rad/s], or a string defining the method for automatic frequency detection:
        "barycenter" frequency is averaged over the power profile, "peak" frequency
        corresponding to the location of the maximum of the on-axis spectrum.
```

File: lasy/profiles/from_insight_file.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        else:
+            assert type(omega0) == float
+
+        # check the complex field convention and correct if needed
+        if omega0 < 0:
+            omega0 *= -1
+            data = np.conj(data)
+            print("Warning: input field will be conjugated")
+
+        # remove the envelope frequencey
```
Comment (MaxThevenet) [inline lasy/profiles/from_insight_file.py]:
```suggestion
        # remove the envelope frequency
```

**Review by MaxThevenet — COMMENTED** [review]
Thanks for this PR! See minor comments below.

**Review by MaxThevenet — APPROVED** [review]

**Pull request merged**